pyccel/codegen/printing/ccode.py

```diff
@@ -1349,9 +1349,22 @@ def _print_For(self, expr):
             step  = self._print(expr.iterable.step )
         else:
             raise NotImplementedError("Only iterable currently supported is Range")
-        return ('for ({target} = {start}; {target} < {stop}; {target} += '
+
+        test_step = expr.iterable.step
+        if isinstance(test_step, PyccelUnarySub):
+            test_step = expr.iterable.step.args[0]
```

Comment:
Why not use an if/else block rather than setting `test_step` twice?

Reply:
just fixed it in [cec0ecc](https://github.com/pyccel/pyccel/pull/780/commits/cec0ecc4ab3d036b396f014715bbe158e3f12a44)

Reply:
Ah sorry I didn't notice that `test_step` was used in the if condition

Reply:
@abkhaoula What Emily meant with her last message is that your original code was correct. Even after b4350b7 the behavior is wrong because you set `test_step` to `None` and then check its type, which does not make sense. So you should change the code back to what it was after commit 29128dc.

Reply:
This is one of those cases where it is OK to set a variable twice! :wink:

Reply:
right make sense. thank you 😅

Reply:
You are welcome!